### PR TITLE
EDGPATRON-114:Support Consortium Authentication & Routing

### PR DESCRIPTION
**Pull request by nirmalr-shah**
[EDGPATRON-114](https://issues.folio.org/browse/EDGPATRON-114): Support Consortium Authentication & Routing

###Testing Evidence
**Scenario-1)**
apiKey : eyJzIjoiMEJ0WUJzODBUQiIsInQiOiJjb25zb3J0aXVtIiwidSI6ImNvbnNvcnRpdW1fYWRtaW4ifQ== (consortium)
alternateTenanaId : consortium
User with external id : 54321 from tenant consortium_admin
image

**_Scenario-2)_**
apiKey : eyJzIjoiMEJ0WUJzODBUQiIsInQiOiJjb25zb3J0aXVtIiwidSI6ImNvbnNvcnRpdW1fYWRtaW4ifQ== (consortium)
alternateTenanaId : consortium
User with external id : 9876 from tenant college_admin(tenant : college)
image

**_Scenario-3)_**
apiKey : eyJzIjoiMEJ0WUJzODBUQiIsInQiOiJjb25zb3J0aXVtIiwidSI6ImNvbnNvcnRpdW1fYWRtaW4ifQ== (consortium)
alternateTenanaId : college
User with external id : 9876 from tenant college_admin(tenant : college)
image

**_Scenario-4)_** W/o alternateTenantId
image

**_Scenario-5)_** /patron/account/:patronId/item/:itemId/renew
image

### CODE REVIEW

**Comment by nirmalr-shah** [comment]
closing because still edge-common PR needs to be merged!!

**Comment by nirmalr-shah** [comment]
Re-opening the PR, as edge-common snapshot change is available on nexus with latest OkapiTenant changes.

File: src/main/resources/ephemeral.properties [hunk 1]
```diff
@@ -10,4 +10,5 @@ tenants=fs00000000,diku
 #######################################################
 
 fs00000000=fs00000000,{FS00000000_IU_PASSWORD}
-diku=diku,{DIKU_IU_PASSWORD}
+diku=diku_admin,{DIKU_IU_PASSWORD}
+consortium=consortium_admin_user,admin
```
Comment (gurleenkaurbp) [inline src/main/resources/ephemeral.properties:14]: The password is hardcoded here, shouldnt it follow the password trend as in the case of diku ?

File: src/main/resources/ephemeral.properties [hunk 2]
```diff
@@ -10,4 +10,5 @@ tenants=fs00000000,diku
 #######################################################
 
 fs00000000=fs00000000,{FS00000000_IU_PASSWORD}
-diku=diku,{DIKU_IU_PASSWORD}
+diku=diku_admin,{DIKU_IU_PASSWORD}
```
Comment (gurleenkaurbp) [inline src/main/resources/ephemeral.properties:13]: Please revert this line if not needed to be changed

File: src/test/java/org/folio/edge/patron/MainVerticleTest.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public class MainVerticleTest {
   private static final String itemId = UUID.randomUUID().toString();
   private static final String instanceId = UUID.randomUUID().toString();
   private static final String holdId = UUID.randomUUID().toString();
-  private static final String apiKey = ApiKeyUtils.generateApiKey(10, "diku", "diku");
+  private static final String apiKey = ApiKeyUtils.generateApiKey(10, "consortium", "consortium_admin_user");
```
Comment (gurleenkaurbp) [inline src/test/java/org/folio/edge/patron/MainVerticleTest.java:80]: Please check if there is a need for a different apiKey for consortium as previously this apiKey was initialized for diku.

File: src/main/resources/ephemeral.properties (hunk 1 above)
Comment (nirmalr-shah) [reply src/main/resources/ephemeral.properties:14]: Actually, after applying the patch it was :

image

But I kept it as {DIKU_IU_PASSWORD} for diku password.

Should I change it to earlier one?

**Review by alexanderkurash — APPROVED** [review]

File: src/main/resources/ephemeral.properties (hunk 2 above)
Comment (nirmalr-shah) [reply src/main/resources/ephemeral.properties]: reverted

File: src/test/java/org/folio/edge/patron/MainVerticleTest.java (hunk 3 above)
Comment (nirmalr-shah) [reply src/test/java/org/folio/edge/patron/MainVerticleTest.java]: reverted with earlier entry of diku

**Review by pavankumar181 — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Review by okolawole-ebsco — APPROVED** [review]

**Pull request merged**